File: examples/01_simple_virtual_machine/python/__main__.py
import pulumi
from typing import Sequence
from pulumiverse_esxi_native import VirtualMachine, NetworkInterfaceArgs

vm = VirtualMachine("vm-test",
                    disk_store="nvme-ssd-datastore",
                    network_interfaces=Sequence[NetworkInterfaceArgs(
                        virtual_network="default"
                    )])

pulumi.export("id", vm.id)
pulumi.export("name", vm.name)
pulumi.export("os", vm.os)

File: examples/01_simple_virtual_machine/go/main.go
package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumiverse/pulumi-esxi-native/sdk/go/esxi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		vm, err := esxi.NewVirtualMachine(ctx, "vm-test", &esxi.VirtualMachineArgs{
			DiskStore: pulumi.String("nvme-ssd-datastore"),
			NetworkInterfaces: esxi.NetworkInterfaceArray{
				esxi.NetworkInterfaceArgs{
					VirtualNetwork: pulumi.String("default"),
				},
			},
		})
		if err != nil {
			return err
		}

		ctx.Export("id", vm.ID())
		ctx.Export("name", vm.Name)
		ctx.Export("os", vm.Os)
		return nil
	})
}
